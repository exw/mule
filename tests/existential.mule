(fn x. 4) : exist t. t -> t
