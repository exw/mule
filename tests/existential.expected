
Desugared:

	(App
	 ((WithType (Quant (() Exist t Unknown (Fn (() (Var (() t)) (Var (() t)))))))
	  (Lam (x (Integer 4)))))

inferred type:

	(Quant (() Exist a Unknown (Fn (() (Var (() a)) (Var (() a))))))

Runtime term:

	(Lam (0 () (Integer 4)))

Evaluated:

	(Lam (0 () (Integer 4)))
